include/os/macos/spl/sys/vnode.h

```diff
@@ -0,0 +1,317 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ *
+ * Copyright (C) 2008 MacZFS
+ * Copyright (C) 2013 Jorgen Lundman <lundman@lundman.net>
+ *
+ */
+
+#ifndef _SPL_VNODE_H
+#define _SPL_VNODE_H
+
+#include <sys/fcntl.h>
+
+#include <sys/mount.h>
+#include <sys/kmem.h>
+#include <sys/mutex.h>
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/uio.h>
+#include <sys/cred.h>
+#include <sys/ubc.h>
+
+#include <kern/locks.h>
+
+
+// Be aware that Apple defines "typedef struct vnode *vnode_t" and
+// ZFS uses "typedef struct vnode vnode_t".
+#undef uio_t
+#undef vnode_t
+#include_next <sys/vnode.h>
+#define vnode_t struct vnode
+#define uio_t struct uio
+
```

Comment:
This is probably the ugliest hack we have to do when trying to fit in with xnu kernel.